### ISSUE TITLE
feature: converting between nativets and deno

### ISSUE THREAD

**Issue by some-user123** [windmill-labs/windmill#2081]
There seems to be a difference between a deno script (created as deno script) and an REST action, that looks very much like a deno script without any import.

It would be nice, if a developer could migrate an REST action to a deno script (to allow imports) and vice versa (to benefit from the performance gain if there are no imports).

**Comment by rubenfiszel** [comment]
You are right about the difference between the 2. The REST language is actually using a deno runtime in process for now just supporting the fetch extension. What I am not sure to understand is if it's not sufficient to copy-paste for now between the 2?

**Comment by some-user123** [comment]
You are absolutely right, that copy&pasting is a suitable workaround here. Changing the type of the action in the `wmill sync` export and pushing it again, would be another possible option.

For me it is a question of UX/DX, especially for new windmill users. One starts with a RESTful call. Then notices the need to validate the JSON structure, parse the CSV, ... whatever. Would be nice to just have a toggle to enable imports.

In the end, an even better solution would be that the action type (deno vs. REST) is determined automatically. Seems like a doable job for windmill to chose the best execution mode for a deno script... and then just have a small info whether this script runs in an optimized runtime.

Just as a side node, it might also make sense to rename REST actions to something else (incl. the logo). Suggestion: "Deno light"
The term "REST" might lead to confusion, as the action type actually has very little to do with RESTful APIs. It suitable for any deno script without imports. Doesn't matter if you transform the result of a postgres action or actually fetch data from an API (be it RESTful or not). And even if you want do RESTful API calls, but use your favorite library, you have to chose a different type.

**Comment by snowyfox** [comment]
I recently just learned that the REST type could be used for more than just REST calls. I had assumed it was more limited due to how it was named.

I have also been working on migrating a bunch of our scripts from Deno to REST types, and one of the UI/UX annoyances I've been dealing with is changing the type of the script in the settings completely wipes out the script I have and replaces it with the bare boilerplate for the new script type. Its a minor inconvenience, as I can just copy / paste the original script back, but is a pain if I forget to copy the script first before trying to switch the type out.

**Comment by rubenfiszel** [comment]
@snowyfox we will fix that. Also you will be happy to hear we can likely improve the concurrency of REST/nativets scripts by at least 20x pretty soon :)
